### ISSUE TITLE
Fetch details of user when adding a contact

### ISSUE THREAD

**Issue by cammellos** [status-im/status-react#14479]
As a user
When I add a contact from the home screen
I want to see their details before I add them to my contacts
So I can recognize them

### Description

When I type a valid key or valid ENS name, we should try to retrieve information about the user from waku and display them.
This will require fetching info from store nodes through status-go.
We should fetch at least display name & profile picture.

Out of scope:
- No loading state required (while we are fetching info, no feedback is given to the user) Not in designs, TBD
- No fallback (if no info are found, we don't display anything) not in designs TBD
- User's profile page

Depends on: https://github.com/status-im/status-mobile/issues/14477
